src/preprocessing/passes/bv_to_int.h

```diff
@@ -0,0 +1,263 @@
+   * and f(x,y) = x && y.
+   * Denote by ITE(a,b) the term that corresponds to the following table:
+   * a | b |  ITE(a,b)
+   * ----------------
+   * 0 | 0 | 0
+   * 0 | 1 | 0
+   * 0 | 2 | 0
+   * 0 | 3 | 0
+   * 1 | 0 | 0
+   * 1 | 1 | 1
+   * 1 | 2 | 0
+   * 1 | 3 | 1
+   * 2 | 0 | 0
+   * 2 | 1 | 0
+   * 2 | 2 | 2
+   * 2 | 3 | 2
+   * 3 | 0 | 0
+   * 3 | 1 | 1
+   * 3 | 2 | 2
+   * 3 | 3 | 3
+   *
+   * (for example, 2 in binary is 10 and 1 in binary is 01, and so doing
+   * "bitwise f" on them gives 00).
+   * The result of this function would be:
+   * ITE(x[1:0], y[1:0])*2^0 + ITE(x[3:2], y[3:2])*2^2
+   */
+  Node createBitwiseNode(Node x,
+                         Node y,
+                         uint64_t bvsize,
+                         uint64_t granularity,
+                         bool (*f)(bool, bool));
+
+  /**
+   * A helper function for createBitwiseNode
+   * - x and y are integer nodes that correspond to the original bit-vector
+   * nodes.
+   * - granularity represents the bitwidth of the original bit-vector nodes.
+   * - table represents a function from pairs of integers to integers.
+   *   The domain of this function consists of pairs of
+   *   integers between 0 (inclusive) and 2^{bitwidth} (exclusive).
+   * The returned node is an ite term that represents this table.
+   */
+  Node createITEFromTable(
+      Node x,
+      Node y,
+      uint64_t granularity,
+      std::map<std::pair<uint64_t, uint64_t>, uint64_t> table);
+
+  /**
+   * A generic function that creates a logical shift node (either left or
+   * right). a << b gets translated to a * 2^b mod 2^k, where k is the bit
+   * width. a >> b gets translated to a div 2^b mod 2^k, where k is the bit
+   * width. The exponentiation operation is translated to an ite for possible
+   * values of the exponent, from 0 to k-1.
+   * If the right operand of the shift is greater than k-1,
+   * the result is 0.
+   * - children: the two operands for the shift
+   * - bvsize: the original bit widths of the operands
+   *   (before translation to integers)
+   * - isLeftShift: true iff the desired operation is a left shift.
+   *
+   */
+  Node createShiftNode(vector<Node> children,
+                       uint64_t bvsize,
+                       bool isLeftShift);
+
+  /**
+   * Returns a node that represents the bitwise negation of n.
+   */
+  Node createBVNotNode(Node n, uint64_t bvsize);
+
+  /**
+   * n is a bit-vector term or formula.
+   * The result is an integer term and is computed
+   * according to the translation specified above.
+   */
+  Node bvToInt(Node n);
+
+  /**
+   * Whenever we introduce an integer variable that represents a bit-vector
+   * variable, we need to guard the range of the newly introduced variable.
+   * For bit width k, the constraint is 0 <= newVar < 2^k.
+   * - newVar is the newly introduced integer variable
+   * - k is the bit width of the original bit-vector variable.
```

Comment:
Try to use consistent formatting for how you describe the parameters, I like how you did it above with `param: description`.

Reply:
This applies to the whole file, will not mark every instance.